tools/faust2clap/faust2clap.py

```diff
@@ -242,47 +275,334 @@ def detect_dsp_type(dsp_path, metadata):
     print("[!] faust compilation failed.")
     sys.exit(1)
 
-#run cMake with macOS sdk path to find stdlib headers
-try:
-    print("[*] running cmake build")
-    sdk_path = subprocess.check_output(["xcrun", "--sdk", "macosx", "--show-sdk-path"], universal_newlines=True).strip()
-    cxx_include_path = "/Library/Developer/CommandLineTools/usr/include/c++/v1"
-
-    # Find the directory containing CMakeLists.txt
-    cmake_dir = this_dir
-    if not os.path.isfile(os.path.join(cmake_dir, "CMakeLists.txt")):
-        # If CMakeLists.txt not in script dir, check if we're in a system installation
-        # In that case, the files should be in the source tree
-        if arch_path and "tools/faust2clap" in os.path.dirname(arch_path):
-            # arch_path points to source tree, use that
-            cmake_dir = os.path.join(os.path.dirname(arch_path), "../../tools/faust2clap")
-            cmake_dir = os.path.abspath(cmake_dir)
+# detect environment and setup build system
+def detect_faust_sdk():
+    """Detect if we're in development or system installation and find SDK components"""
     
-    if not os.path.isfile(os.path.join(cmake_dir, "CMakeLists.txt")):
-        print("[!] CMakeLists.txt not found. faust2clap installation may be incomplete.")
-        print(f"[!] Searched in: {cmake_dir}")
-        sys.exit(1)
+    # check for development environment (has external dependencies)
+    dev_clap_sdk = os.path.join(this_dir, "../../external/clap-sdk")
+    dev_clap_helpers = os.path.join(this_dir, "../../external/clap-helpers")
     
-    subprocess.run([
-        "cmake", "-S", cmake_dir, "-B", os.path.join(cmake_dir, "build"),
-        f"-DCMAKE_CXX_FLAGS=-isysroot {sdk_path} -I{cxx_include_path}"
-    ], check=True, capture_output=True)
-
-    subprocess.run(["cmake", "--build", os.path.join(cmake_dir, "build")], check=True, capture_output=True)
-
-    print("[✓] build completed successfully.")
-except subprocess.CalledProcessError as e:
-    print("[!] cmake build failed.")
-    print(f"[stderr]\n{e.stderr.decode() if e.stderr else 'No stderr'}")
-    print(f"[stdout]\n{e.stdout.decode() if e.stdout else 'No stdout'}")
-    sys.exit(1)
+    if os.path.isdir(dev_clap_sdk) and os.path.isdir(dev_clap_helpers):
+        return {
+            'type': 'development',
+            'clap_sdk': dev_clap_sdk,
+            'clap_helpers': dev_clap_helpers,
+            'faust_lib': os.path.join(this_dir, "../.."),
+            'cmake_template': os.path.join(this_dir, "CMakeLists.txt"),
+            'plugin_template': os.path.join(this_dir, "lib/plugin.cc"),
+            'gui_glue': os.path.join(this_dir, "lib/faust_gui_glue.cpp"),
+            'plist_template': os.path.join(this_dir, "cmake/generic.plist.in")
+        }
+    
+    # check for system installation SDK
+    # try multiple system paths where SDK might be installed
+    system_paths = [
+        "/usr/local/share/faust/tools/faust2clap",
```

Comment:
This` /usr/local/share/faust/tools/faust2clap`  kind of patch should better be built using "faust -archdir" command which is here exactly for that: discover where the installed Faust version is !